public/app/features/explore/state/query.ts

```diff
@@ -231,15 +236,19 @@ export function cancelQueries(exploreId: ExploreId): ThunkResult<void> {
   return (dispatch, getState) => {
     dispatch(scanStopAction({ exploreId }));
     dispatch(cancelQueriesAction({ exploreId }));
-    dispatch(
-      storeLogsVolumeDataProviderAction({
-        exploreId,
-        logsVolumeDataProvider: undefined,
-      })
-    );
-    // clear any incomplete data
-    if (getState().explore[exploreId]!.logsVolumeData?.state !== LoadingState.Done) {
-      dispatch(cleanLogsVolumeAction({ exploreId }));
+
+    const suppQueries = getState().explore[exploreId]!.suppQueries;
+    // Cancel all data providers
+    for (const type of Object.keys(suppQueries)) {
```

Comment:
What I mean here https://github.com/grafana/grafana/pull/60998/files#r1062676102 is.

```javaScript
export enum SuppQueryType {
  LogsVolume = 'LogsVolume',
}
const TYPES: SuppQueryType[] = [SuppQueryType.LogsVolume];
```

And thsi code here becomes:

```suggestion
    for (const type of TYPES) {
       // And we avoid both Object.x() functions and code just to satisfy TS
    }
```

But that's just me. Feel free to go either direction you're more comfortable with.

Reply:
As mentioned updated to https://github.com/grafana/grafana/commit/2c7b92ea0ec7a9f14049161befe6ad5ca8824f53 to use const type of SUPPORTING_QUERY_TYPES.